### PR TITLE
Improve VS Code intellisense

### PR DESCRIPTION
**Pull request by gtaharaedmonds**
### Summary

Use VS Code's CMake Tools extension to get much better (on my machine at least) syntax highlighting, autocomplete, and navigation features.

To enable:
1. Install [CMake Tools](https://marketplace.visualstudio.com/items?itemName=ms-vscode.cmake-tools).
2. Press `Ctrl+P` and run `>CMake: Select a Kit` and select `[Unspecified]`.
3. For good intellisense for embedded targets, press `Ctrl+P` and run `>CMake: Select Variant` and select `Embedded`.
4. For good intellisense for tests press `Ctrl+P` and run `>CMake: Select Variant` and select `Tests`.
5. Build the binaries.
6. Reload your window: `Ctrl+P` and `>Developer: Reload Window`.

With the `Embedded` variant active the support is pretty good when writing tests, and vice-versa. The experience is much better however when you have the right variant selected. Make sure to reload your window after changing the active variant.

### Changelist 

A few config files in the `.vscode` directory.

### Testing Done

- [x] Messing around on my setup 

### Checklist
*Please change `[ ]` to `[x]` when you are ready.*
- [x] I have read and followed the code conventions detailed in [README.md](../README.md) (*This will save time for both you and the reviewer!*).
- [x] If this pull request is longer then **500** lines, I have provided *explicit* justification in the summary above explaining why I *cannot* break this up into multiple pull requests (*Small PR's are faster and less painful for everyone involved!*).

### CODE REVIEW

**Review by philam2001 — APPROVED** [review]

File: .vscode/c_cpp_properties.json [hunk 1]
```diff
@@ -0,0 +1,9 @@
+{
+    "configurations": [
+        {
+            "name": "CMake",
+            "configurationProvider": "ms-vscode.cmake-tools"
+        }
+    ],
+    "version": 4
+}
```
Comment (will-chaba) [inline .vscode/c_cpp_properties.json:9]: nit: missing newline
```suggestion
}

```

File: .vscode/cmake-variants.yaml [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+            env:
+                PLATFORM: arm
+            buildType: Debug
+        x86: 
+            short: Tests
+            settings:
+                PLATFORM: x86
+            env:
+                PLATFORM: x86
+            buildType: Debug
```
Comment (will-chaba) [inline .vscode/cmake-variants.yaml:17]: nit: missing newline
```suggestion
            buildType: Debug
            
```

File: .vscode/settings.json [hunk 3]
```diff
@@ -6,5 +6,8 @@
 			],
 			"url": "/can_bus/json/schema_tx.json"
 		}
-	]
+	],
+	"cmake.buildDirectory": "${workspaceFolder}/build_${env:PLATFORM}",
+	"cmake.configureOnEdit": false,
+	"cmake.configureOnOpen": true
 }
```
Comment (will-chaba) [inline .vscode/settings.json:13]: nit: missing newline
```suggestion
}

```

**Review by will-chaba — APPROVED** [review]
nit: missing newlines

**Pull request merged**